hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/AbstractFindTargetGreedy.java

```diff
@@ -185,10 +192,26 @@ private boolean canSizeEnterTarget(DatanodeDetails target, long size) {
       // MaxSizeEnteringTarget
       //2 current usage of target datanode plus sizeEnteringAfterMove
       // is smaller than or equal to upperLimit
-      return sizeEnteringAfterMove <= config.getMaxSizeEnteringTarget() &&
-          Double.compare(nodeManager.getUsageInfo(target)
-              .calculateUtilization(sizeEnteringAfterMove), upperLimit) <= 0;
+      if (sizeEnteringAfterMove > config.getMaxSizeEnteringTarget()) {
+        logger.debug("{} bytes cannot enter datanode {} because 'size" +
+                ".entering.target.max' limit is {} and {} bytes have already " +
+                "entered.", size, target.getUuidString(),
+            config.getMaxSizeEnteringTarget(),
+            sizeEnteringNode.get(target));
+        return false;
+      }
+      if (Double.compare(nodeManager.getUsageInfo(target)
+          .calculateUtilization(sizeEnteringAfterMove), upperLimit) > 0) {
+        logger.debug("{} bytes cannot enter datanode {} because its " +
+                "utilization will exceed the upper limit of {}.", size,
+            target.getUuidString(), upperLimit);
+        return false;
+      }
+      return true;
     }
+
+    logger.warn("No record of how much size has entered datanode {}",
```

Comment:
Is this `warn` worth? or should should it be a `debug` or `info`?

Reply:
I don't have a strong opinion on this. When searching logs, if this warning shows up then I know something is wrong. It can be `info` too, but `debug` won't be much help.